components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java

```diff
@@ -161,6 +161,58 @@ public static void createApplication(String appName, String appOwner, String app
                 .createApplication(serviceProvider, tenantDomain, appOwner);
     }
 
+    /**
+     * Create portal SaaS application.
+     *
+     * @param appName        Application name.
+     * @param appOwner       Application owner.
+     * @param appDescription Application description.
+     * @param consumerKey    Consumer key.
+     * @param consumerSecret Consumer secret.
+     * @param portalPath     Portal path.
+     * @throws IdentityApplicationManagementException IdentityApplicationManagementException.
+     */
+    public static void createApplication(String appName, String appOwner, String appDescription, String consumerKey,
+                                         String consumerSecret, String tenantDomain, String portalPath)
+            throws IdentityApplicationManagementException {
```

Comment:
can you remove the body of the previous `createApplication(String appName, String appOwner, String appDescription, String consumerKey,
            String consumerSecret, String tenantDomain)` and call the new method?

having two methods with similar logic introduces redundancy

Reply:
+1 Call the overloaded method internally

Reply:
resolved

Reply:
let's add a deprecated annotation for the method `createApplication(String appName, String appOwner, String appDescription, String consumerKey, String consumerSecret, String tenantDomain)`

Reply:
resolved